PWGHF/TableProducer/candidateSelectorOmegac0ToOmegaPi.cxx

```diff
@@ -160,409 +182,494 @@ struct HfCandidateSelectorToOmegaPi {
-
-      auto trackPiFromLam = trackV0NegDau;
-      auto trackPrFromLam = trackV0PosDau;
-
-      int8_t signDecay = candidate.signDecay(); // sign of pi <- cascade
-
-      if (signDecay > 0) {
-        trackPiFromLam = trackV0PosDau;
-        trackPrFromLam = trackV0NegDau;
-        registry.fill(HIST("hSelSignDec"), 1); // anti-particle decay
-      } else if (signDecay < 0) {
-        registry.fill(HIST("hSelSignDec"), 0); // particle decay
-      }
-
-      // eta selection
-      double etaV0PosDau = candidate.etaV0PosDau();
-      double etaV0NegDau = candidate.etaV0NegDau();
-      double etaKaFromCasc = candidate.etaBachFromCasc();
-      double etaPiFromCharmBaryon = candidate.etaBachFromCharmBaryon();
-      if (std::abs(etaV0PosDau) > etaTrackLFDauMax) {
-        resultSelections = false;
-        registry.fill(HIST("hSelEtaPosV0Dau"), 0);
-      } else {
-        registry.fill(HIST("hSelEtaPosV0Dau"), 1);
-      }
-      if (std::abs(etaV0NegDau) > etaTrackLFDauMax) {
-        resultSelections = false;
-        registry.fill(HIST("hSelEtaNegV0Dau"), 0);
-      } else {
-        registry.fill(HIST("hSelEtaNegV0Dau"), 1);
-      }
-      if (std::abs(etaKaFromCasc) > etaTrackLFDauMax) {
-        resultSelections = false;
-        registry.fill(HIST("hSelEtaKaFromCasc"), 0);
-      } else {
-        registry.fill(HIST("hSelEtaKaFromCasc"), 1);
-      }
-      if (std::abs(etaPiFromCharmBaryon) > etaTrackCharmBachMax) {
-        resultSelections = false;
-        registry.fill(HIST("hSelEtaPiFromCharm"), 0);
-      } else {
-        registry.fill(HIST("hSelEtaPiFromCharm"), 1);
-      }
-
-      // minimum radius cut (LFcut)
-      if (RecoDecay::sqrtSumOfSquares(candidate.xDecayVtxCascade(), candidate.yDecayVtxCascade()) < radiusCascMin) {
-        resultSelections = false;
-        registry.fill(HIST("hSelRadCasc"), 0);
-      } else {
-        registry.fill(HIST("hSelRadCasc"), 1);
-      }
-      if (RecoDecay::sqrtSumOfSquares(candidate.xDecayVtxV0(), candidate.yDecayVtxV0()) < radiusV0Min) {
-        resultSelections = false;
-        registry.fill(HIST("hSelRadV0"), 0);
-      } else {
-        registry.fill(HIST("hSelRadV0"), 1);
-      }
-
-      // cosPA (LFcut)
-      if (candidate.cosPACasc() < cosPACascMin) {
-        resultSelections = false;
-        registry.fill(HIST("hSelCosPACasc"), 0);
-      } else {
-        registry.fill(HIST("hSelCosPACasc"), 1);
-      }
-      if (candidate.cosPAV0() < cosPAV0Min) {
-        resultSelections = false;
-        registry.fill(HIST("hSelCosPAV0"), 0);
-      } else {
-        registry.fill(HIST("hSelCosPAV0"), 1);
-      }
-
-      // cascade and v0 daughters dca cut (LF cut)
-      if (candidate.dcaCascDau() > dcaCascDauMax) {
-        resultSelections = false;
-        registry.fill(HIST("hSelDCACascDau"), 0);
-      } else {
-        registry.fill(HIST("hSelDCACascDau"), 1);
-      }
+    if constexpr (std::is_same_v<CandidateType, aod::HfOmegacKf>) {
```

Comment:
This is not the right place where to check wheter you are consuming the `HfOmegaKf` table or not (see later)